src/createTokenBridge.ts

```diff
@@ -34,6 +35,43 @@ import { isCustomFeeTokenAddress } from './utils/isCustomFeeTokenAddress';
 import { WithTokenBridgeCreatorAddressOverride } from './types/createTokenBridgeTypes';
 import { TransactionRequestGasOverrides } from './utils/gasOverrides';
 import { getBlockExplorerUrl } from './utils/getBlockExplorerUrl';
+import { tokenBridgeCreatorABI } from './contracts/TokenBridgeCreator';
+import { getTokenBridgeCreatorAddress } from './utils';
+
+export async function assertTokenBridgeDoesntExist<
+  TParentChain extends Chain | undefined,
+  TOrbitChain extends Chain | undefined,
+>({
+  parentChainPublicClient,
+  orbitChainPublicClient,
+  tokenBridgeCreatorAddress,
+  rollupAddress,
+}: {
+  parentChainPublicClient: PublicClient<Transport, TParentChain>;
+  orbitChainPublicClient: PublicClient<Transport, TOrbitChain>;
+  tokenBridgeCreatorAddress?: Address;
+  rollupAddress: Address;
+}) {
+  const inbox = await parentChainPublicClient.readContract({
+    address: rollupAddress,
+    abi: parseAbi(['function inbox() view returns (address)']),
+    functionName: 'inbox',
+  });
+
+  const [router] = await parentChainPublicClient.readContract({
+    address: tokenBridgeCreatorAddress ?? getTokenBridgeCreatorAddress(parentChainPublicClient),
+    abi: tokenBridgeCreatorABI,
+    functionName: 'inboxToL2Deployment',
+    args: [inbox],
+  });
+
+  if (router) {
+    const code = await orbitChainPublicClient.getBytecode({ address: router });
+    if (code) {
+      throw new Error('Token bridge was already deployed');
+    }
+  }
+}
```

Comment:
i'm wondering if it's a good idea to log something like `Passed token bridge check. Token bridge hasn't been deployed yet.`